### PR TITLE
feat: Make toon use JSON tags if no TOON tags are present.

### PR DESCRIPTION
**Pull request by one-two-three-four-five-six-seven**
## Description
With these changes if no toon tags are present on a struct the buildStructMeta function inside of `internal/codex/structmeta.go` will fallback to using json tags. This allows external packages structs that contain json tags and not toon tags to be used with proper field naming. I believe this is a good change considering the main use of toon is just converting JSON -> TOON to save on tokens. It technicaly can be a breaking change if people are not using toon tags, however considering this is being passed to an LLM (i hope) this isnt too much of an issue.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Test coverage improvement

## Changes Made

- Toon will use JSON tags if TOON tags are not present

## SPEC Compliance

- [ ] This PR implements/fixes spec compliance
- [ ] Spec section(s) affected:
- [ ] Spec version:

## Testing

I added a test that checks the priority of toon and json tags works as expected.
- [X] All existing tests pass
- [X] Added new tests for changes
- [X] Tests cover edge cases and spec compliance 

## Pre-submission Checklist

- [X] My code follows the project's coding standards
- [X] I have run code formatting/linting tools
- [X] I have added tests that prove my fix/feature works
- [X] New and existing tests pass locally 
- [X] I have updated documentation if needed
- [ ] I have reviewed the [TOON specification](https://github.com/toon-format/spec) for relevant sections

## Breaking Changes

- [ ] No breaking changes
- [X] Breaking changes (describe migration path below)

If for whatever reason you are using the default struct field name fallback for toon and have json tags you will need to add toon tags.

```go
type Example struct {
	Foo string `json:"foo"`
	Bar int    `json:"bar"`
}
```
now becomes
```go
type Example struct {
	Foo string `json:"foo" toon:"foo"`
	Bar int    `json:"bar" toon:"bar"`
}
```

type str

### CODE REVIEW

File: internal/codec/structmeta.go [hunk 1]
```diff
@@ -37,9 +37,15 @@ func buildStructMeta(t reflect.Type) structMeta {
 			continue
 		}
 		tag := sf.Tag.Get("toon")
+		// Only check the json tag value if toon is undefined
+		// I don't think its worth adding an option or anything considering you can just overide this with actually setting a toon tag.
+		if tag == "" {
+			tag = sf.Tag.Get("json")
+		}
```
Comment (ccoVeille) [inline internal/codec/structmeta.go]: I would control this via an option. I'm unsure anyone want an automatic fallback on json field

Comment (one-two-three-four-five-six-seven) [reply internal/codec/structmeta.go]: Done

File: api.go [hunk 2]
```diff
@@ -87,6 +87,11 @@ func WithTimeFormatter(formatter func(time.Time) string) EncoderOption {
 	return codec.WithTimeFormatter(formatter)
 }
 
+// WithJsonFallbackEncoder enables using json tags if there is a lack of toon tags (Encoder is used in the name since this is present on encoding / decoding)
+func WithJsonFallbackEncoder(enabled bool) EncoderOption {
+	return codec.WithJsonFallbackEncoder(enabled)
+}
```
Comment (ccoVeille) [inline api.go]: By convention, initialism and acronym are always in upper case. So please consider this


```suggestion
// WithJSONFallbackEncoder enables using json tags if there is a lack of toon tags (Encoder is used in the name since this is present on encoding / decoding)
func WithJSONFallbackEncoder(enabled bool) EncoderOption {
	return codec.WithJSONFallbackEncoder(enabled)
}
```

File: api.go [hunk 3]
```diff
@@ -123,6 +128,11 @@ func WithDecoderDocumentDelimiter(delimiter Delimiter) DecoderOption {
 	return codec.WithDecoderDocumentDelimiter(delimiter)
 }
 
+// WithJsonFallbackDecoder enables using json tags if there is a lack of toon tags (Decoder is used in the name since this is present on encoding / decoding)
+func WithJsonFallbackDecoder(enabled bool) DecoderOption {
+	return codec.WithJsonFallbackDecoder(enabled)
+}
```
Comment (ccoVeille) [inline api.go:134]: See

https://github.com/toon-format/toon-go/pull/16#discussion_r3070018484


```suggestion
// WithJSONFallbackDecoder enables using json tags if there is a lack of toon tags (Decoder is used in the name since this is present on encoding / decoding)
func WithJSONFallbackDecoder(enabled bool) DecoderOption {
	return codec.WithJSONFallbackDecoder(enabled)
}
```

File: internal/codec/options.go [hunk 4]
```diff
@@ -65,6 +66,14 @@ func defaultEncoderOptions() encoderOptions {
 	}
 }
 
+// WithJsonFallbackEncoder enables using json tags if there is a lack of toon tags
+// (Encoder is used in the name since this is present on encoding / decoding)
+func WithJsonFallbackEncoder(fallback bool) EncoderOption {
```
Comment (ccoVeille) [inline internal/codec/options.go]: See https://github.com/toon-format/toon-go/pull/16#discussion_r3070018484

```suggestion
// WithJSONFallbackEncoder enables using json tags if there is a lack of toon tags
// (Encoder is used in the name since this is present on encoding / decoding)
func WithJsonFallbackEncoder(fallback bool) EncoderOption {
```

Comment (ccoVeille) [reply internal/codec/options.go]: This one and a few other still mention `Json` and not `JSON`
